pkg/adapter/config/encryptionAdapter.go

```diff
@@ -2,29 +2,132 @@ package config
 
 import (
 	"context"
+	"errors"
+	"github.com/cloudogu/cesapp-lib/keys"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"github.com/cloudogu/k8s-dogu-operator/controllers/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type PublicKeyConfigEncryptionAdapter struct {
+	secrets   secret
+	registry  etcdRegistry
+	namespace string
 }
 
-func NewPublicKeyConfigEncryptionAdapter() *PublicKeyConfigEncryptionAdapter {
-	return &PublicKeyConfigEncryptionAdapter{}
+func NewPublicKeyConfigEncryptionAdapter(secretClient secret, registry etcdRegistry, namespace string) *PublicKeyConfigEncryptionAdapter {
+	return &PublicKeyConfigEncryptionAdapter{secrets: secretClient, registry: registry, namespace: namespace}
 }
 
 func (p PublicKeyConfigEncryptionAdapter) Encrypt(
 	ctx context.Context,
 	name common.SimpleDoguName,
 	value common.SensitiveDoguConfigValue,
 ) (common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	pubkey, err := resource.GetPublicKey(p.registry, string(name))
+	if err != nil {
+		return "", &domainservice.NotFoundError{
+			WrappedError: err,
+			Message:      "could not get public key",
+		}
+	}
+	encryptedValue, err := pubkey.Encrypt(string(value))
+	if err != nil {
+		return "", domainservice.NewInternalError(err, "could not encrypt value")
+	}
+	return common.EncryptedDoguConfigValue(encryptedValue), nil
 }
 
 func (p PublicKeyConfigEncryptionAdapter) EncryptAll(
 	ctx context.Context,
 	entries map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue,
 ) (map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	encryptedEntries := map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue{}
+	var encryptionErrors []error
+	pubkeys := map[string]*keys.PublicKey{}
+	for configKey, configValue := range entries {
+		doguname := string(configKey.DoguName)
+		pubkey, pubkeyKnown := pubkeys[doguname]
+		if !pubkeyKnown {
+			pubkeyFromRegistry, err := resource.GetPublicKey(p.registry, doguname)
+			if err != nil {
+				encryptionErrors = append(encryptionErrors, domainservice.NewNotFoundError(err, "could not get public key for dogu %v", doguname))
+				continue
+			} else {
+				pubkey = pubkeyFromRegistry
+				pubkeys[doguname] = pubkeyFromRegistry
+			}
+		}
+		encryptedValue, err := pubkey.Encrypt(string(configValue))
+		if err != nil {
+			encryptionErrors = append(encryptionErrors, domainservice.NewInternalError(err, "could not encrypt value for dogu %v", doguname))
+		}
+		encryptedEntries[configKey] = common.EncryptedDoguConfigValue(encryptedValue)
+	}
+	return encryptedEntries, errors.Join(encryptionErrors...)
```

Comment:
This is very deeply nested. Maybe consider to load all pubKeys beforehand in a separate helper function for the given config keys.